File: app/src/main/java/com/gs/buluo/store/view/activity/WalletActivity.java
package com.gs.buluo.store.view.activity;

import android.content.Context;
import android.content.DialogInterface;
import android.content.Intent;
import android.os.Bundle;
import android.text.TextUtils;
import android.view.View;
import android.widget.TextView;

import com.gs.buluo.store.Constant;
import com.gs.buluo.store.R;
import com.gs.buluo.store.TribeApplication;
import com.gs.buluo.store.bean.WalletAccount;
import com.gs.buluo.store.presenter.BasePresenter;
import com.gs.buluo.store.presenter.WalletPresenter;
import com.gs.buluo.common.utils.ToastUtils;
import com.gs.buluo.store.view.impl.IWalletView;

import butterknife.Bind;

/**
 * Created by hjn on 2016/11/17.
 */
public class WalletActivity extends BaseActivity implements View.OnClickListener, IWalletView, DialogInterface.OnDismissListener {
    @Bind(R.id.wallet_integer)
    TextView mInterger;
    @Bind(R.id.wallet_float)
    TextView mFloat;
    Context mCtx;
    private String pwd;
    private float balance = -1;
    private float withdrawCharge;

    @Override
    protected void bindView(Bundle savedInstanceState) {
        mCtx = this;
        findViewById(R.id.wallet_bill).setOnClickListener(this);
        findViewById(R.id.wallet_card).setOnClickListener(this);
        findViewById(R.id.wallet_financial).setOnClickListener(this);
        findViewById(R.id.wallet_pwd).setOnClickListener(this);
        findViewById(R.id.wallet_cash).setOnClickListener(this);
        findViewById(R.id.wallet_receive).setOnClickListener(this);
    }

    @Override
    protected int getContentLayout() {
        return R.layout.activity_wallet;
    }

    @Override
    protected void onResume() {
        super.onResume();
        showLoadingDialog();
        ((WalletPresenter) mPresenter).getWalletInfo();
    }

    @Override
    public void onClick(View v) {
        if (balance==-1){
            ToastUtils.ToastMessage(getCtx(),R.string.connect_fail);
            return;
        }
        Intent intent = new Intent();
        switch (v.getId()) {
            case R.id.wallet_bill:
                intent.setClass(WalletActivity.this, BillActivity.class);
                startActivity(intent);
                break;
            case R.id.wallet_card:
                intent.setClass(mCtx, BankCardActivity.class);
                startActivity(intent);
                break;
            case R.id.wallet_receive:
                intent.setClass(mCtx,PayCodeActivity.class);
                startActivity(intent);
                break;
            case R.id.wallet_financial:
                ToastUtils.ToastMessage(getCtx(),R.string.no_function);
                break;
            case R.id.wallet_pwd:
                if (TextUtils.isEmpty(pwd)) {
                    intent.setClass(mCtx, UpdateWalletPwdActivity.class);
                } else {
                    intent.putExtra(Constant.WALLET_PWD, pwd);
                    intent.setClass(mCtx, ConfirmActivity.class);
                }
                startActivity(intent);
                break;
            case R.id.wallet_cash:
                if (!TribeApplication.getInstance().isBf_withdraw()){
                    ToastUtils.ToastMessage(getCtx(),R.string.no_function);
                    break;
                }
                intent.putExtra(Constant.WALLET_AMOUNT,balance);
                intent.putExtra(Constant.WALLET_PWD,pwd);
                intent.putExtra(Constant.POUNDAGE,withdrawCharge);
                intent.setClass(getCtx(),CashActivity.class);
                startActivity(intent);
                break;
        }
    }


    @Override
    protected BasePresenter getPresenter() {
        return new WalletPresenter();
    }

    @Override
    public void getWalletInfoFinished(WalletAccount account) {
        pwd = account.password;
        balance = account.balance;
        withdrawCharge = account.withdrawCharge;
        setData(balance+"");
    }

    public void setData(String price) {
        if (price == null) return;
        String[] arrs = price.split("\\.");
        if (arrs.length > 1) {
            mInterger.setText(arrs[0]);
            mFloat.setText(arrs[1]);
        } else {
            mInterger.setText(price);
            mFloat.setText("00");
        }
    }

    @Override
    public void showError(int res) {
        ToastUtils.ToastMessage(this, getString(res));
    }

    @Override
    public void onDismiss(DialogInterface dialog) {
        ((WalletPresenter) mPresenter).getWalletInfo();
    }
}

File: app/src/main/java/com/gs/buluo/store/kotlin/activity/WalletActivity.kt
package com.gs.buluo.store.kotlin.activity

import com.gs.buluo.store.kotlin.presenter.WalletPresenter
import kotlinx.android.synthetic.main.activity_wallet.*

/**
 * Created by hjn on 2016/11/17.
 */
class WalletActivity: KotBaseActivity(), android.view.View.OnClickListener, com.gs.buluo.store.view.impl.IWalletView, android.content.DialogInterface.OnDismissListener {
    override fun bindView(savedInstanceState: android.os.Bundle?) {
        wallet_bill.setOnClickListener(this)
        wallet_card.setOnClickListener(this)
        wallet_financial.setOnClickListener(this)
        wallet_pwd.setOnClickListener(this)
        wallet_cash.setOnClickListener(this)
        wallet_receive.setOnClickListener(this)
        presenter  = WalletPresenter(this)
    }

    override val contentLayout: Int
        get() = com.gs.buluo.store.R.layout.activity_wallet


    private var pwd: String = ""
    private var balance = -1f
    private var withdrawCharge: Float = 0.toFloat()

    override fun onResume() {
        super.onResume()
        showLoadingDialog()
        getData()
    }

    private fun getData() {
        (presenter as WalletPresenter).getWalletInfo()
    }

    override fun onClick(v: android.view.View) {
        if (balance == -1f) {
            com.gs.buluo.common.utils.ToastUtils.ToastMessage(this, com.gs.buluo.store.R.string.connect_fail)
            return
        }
        val intent = android.content.Intent()
        when (v.id) {
            com.gs.buluo.store.R.id.wallet_bill -> {
                intent.setClass(this, com.gs.buluo.store.view.activity.BillActivity::class.java)
                startActivity(intent)
            }
            com.gs.buluo.store.R.id.wallet_card -> {
                intent.setClass(this, com.gs.buluo.store.view.activity.BankCardActivity::class.java)
                startActivity(intent)
            }
            com.gs.buluo.store.R.id.wallet_receive -> {
                intent.setClass(this, com.gs.buluo.store.view.activity.PayCodeActivity::class.java)
                startActivity(intent)
            }
            com.gs.buluo.store.R.id.wallet_financial -> com.gs.buluo.common.utils.ToastUtils.ToastMessage(this, com.gs.buluo.store.R.string.no_function)
            com.gs.buluo.store.R.id.wallet_pwd -> {
                if (android.text.TextUtils.isEmpty(pwd)) {
                    intent.setClass(this, com.gs.buluo.store.view.activity.UpdateWalletPwdActivity::class.java)
                } else {
                    intent.putExtra(com.gs.buluo.store.Constant.WALLET_PWD, pwd)
                    intent.setClass(this, com.gs.buluo.store.view.activity.ConfirmActivity::class.java)
                }
                startActivity(intent)
            }
            com.gs.buluo.store.R.id.wallet_cash -> {
                if (!com.gs.buluo.store.TribeApplication.getInstance().isBf_withdraw) {
                    com.gs.buluo.common.utils.ToastUtils.ToastMessage(this, com.gs.buluo.store.R.string.no_function)
                    return
                }
                intent.putExtra(com.gs.buluo.store.Constant.WALLET_AMOUNT, balance)
                intent.putExtra(com.gs.buluo.store.Constant.WALLET_PWD, pwd)
                intent.putExtra(com.gs.buluo.store.Constant.POUNDAGE, withdrawCharge)
                intent.setClass(this, com.gs.buluo.store.view.activity.CashActivity::class.java)
                startActivity(intent)
            }
        }
    }

    override fun getWalletInfoFinished(account: com.gs.buluo.store.bean.WalletAccount) {
        pwd = account.password
        balance = account.balance
        withdrawCharge = account.withdrawCharge
        setData(balance.toString() + "")

//        var (account ,id ) = account
    }

    fun setData(price: String?) {
        if (price == null) return
        val arrs = price.split("\\.".toRegex()).dropLastWhile { it.isEmpty() }.toTypedArray()
        if (arrs.size > 1) {
            wallet_integer.text = arrs[0]
            wallet_float.text = arrs[1]
        } else {
            wallet_integer.text = price
            wallet_float.text = "00"
        }
    }

    override fun showError(res: Int) {
        com.gs.buluo.common.utils.ToastUtils.ToastMessage(this, getString(res))
    }

    override fun onDismiss(dialog: android.content.DialogInterface) {
        (presenter as WalletPresenter).getWalletInfo()
    }
}
